RiotSharpTest/RiotApiTest.cs

```diff
@@ -135,22 +136,52 @@ public void GetChampionAsync_Test()
 
         [TestMethod]
         [TestCategory("RiotApi")]
-        public void GetMasteryPages_Test()
+        public void GetMasteryPages_ExistingSummonerId_HasMasteryPages()
         {
-            var masteries = api.GetMasteryPages(RiotApiTestBase.summonersRegion, RiotApiTestBase.summonerIds);
+            var pages = api.GetMasteryPages(RiotApiTestBase.summonersRegion, RiotApiTestBase.summoner1Id);
 
-            Assert.IsNotNull(masteries);
-            Assert.AreEqual(RiotApiTestBase.summonerIds.Distinct().Count(), masteries.Count);
+            Assert.IsNotNull(pages);
+            Assert.IsTrue(pages.Count >= 1 && pages.Count <= 20);
+        }
+
+        [TestMethod]
+        [TestCategory("RiotApi")]
+        public void GetMasteryPages_InvalidSummonerId_ThrowsResouceNotFound()
+        {
+            try
+            {
+                api.GetMasteryPages(RiotApiTestBase.summonersRegion, RiotApiTestBase.invalidSummonerId);
+                Assert.Fail();
+            }
+            catch (RiotSharpException e)
+            {
+                Assert.AreEqual(e.HttpStatusCode, HttpStatusCode.NotFound);
+            }
         }
 
         [TestMethod]
         [TestCategory("RiotApi"), TestCategory("Async")]
-        public void GetMasteryPagesAsync_Test()
+        public void GetMasteryPagesAsync_ExistingSummonerId_HasMasteryPages()
         {
-            var masteries = api.GetMasteryPagesAsync(RiotApiTestBase.summonersRegion, RiotApiTestBase.summonerIds);
+            var pages = api.GetMasteryPagesAsync(RiotApiTestBase.summonersRegion, RiotApiTestBase.summoner1Id);
 
-            Assert.IsNotNull(masteries.Result);
-            Assert.AreEqual(RiotApiTestBase.summonerIds.Distinct().Count(), masteries.Result.Count);
+            Assert.IsNotNull(pages.Result);
+            Assert.IsTrue(pages.Result.Count >= 1 && pages.Result.Count <= 20);
+        }
+
+        [TestMethod]
+        [TestCategory("RiotApi"), TestCategory("Async")]
+        public async Task GetMasteryPagesAsync_InvalidSummonerId_ThrowsResouceNotFound()
+        {
+            try
+            {
+                await api.GetMasteryPagesAsync(RiotApiTestBase.summonersRegion, RiotApiTestBase.invalidSummonerId);
+                Assert.Fail();
+            }
+            catch (RiotSharpException e)
+            {
+                Assert.AreEqual(e.HttpStatusCode, HttpStatusCode.NotFound);
+            }
```

Comment:
I'm thinking about changing `HandleAggregateException` to:
```
void HandleAggregateException(AggregateException exception, Action<RiotSharpException> riotSharpExceptionHandler = null)
```

Which allows for something like:
```
HandleAggregateException(exception, (riotException) =>
{
    Assert.AreEqual(riotException.HttpStatusCode, HttpStatusCode.NotFound);  
});
```
By default it would still use `HandleRiotSharpException` for `EnsureCredibility`.

Thoughts? Good idea? Bad idea?